sklego/preprocessing.py

```diff
@@ -197,6 +198,7 @@ def transform(self, X):
         :param X: ``pd.DataFrame`` on which we apply the column selection
         :returns: ``pd.DataFrame`` with only the selected columns
         """
+        self._check_X_for_type(X)
```

Comment:
call _check_X_for_type(X) in the transform method, because this class is dedicated to DF only and checking X type in fit is not sufficient because if we fit a DF but try to transform an array then it throws,
    `IndexError: only integers, slices (:), ellipsis (...), numpy.newaxis (None) and integer 
     or boolean arrays are valid indices`

but the more suitable and correct error would be,
`TypeError: Provided variable X is not of type pandas.DataFrame`